src/fire-event.js

```diff
@@ -11,8 +11,17 @@ Object.keys(domFireEvent).forEach((key) => {
     // we hit the Preact listeners.
     const eventName = `on${key.toLowerCase()}`
     const isInElem = eventName in elem
+    // Preact changes all change events to input events. This is normally handled directly
+    // but when running 'preact/compat' this is done via custom JS which renames the event prop,
+    // making the event name out of sync.
+    // The problematic code is in: preact/compat/src/render.js > handleDomVNode()
+    const keyFiltered = key === 'change' ? 'input' : key
```

Comment:
Preact only aliases change events to input events when `preact/compat` is used, to match React's behavior.

Aliasing as you have here, unconditionally, breaks functionality for anyone not using `preact/compat`, and therefore expecting the native change event. Need some way to apply this only when `preact/compat` is in use.

Reply:
You are absolutely correct, I misunderstood this in their documentation. I will look for a possible way to do this and update my pull request. Thanks for the review!

Reply:
I am thinking to apply the check if `compat` is used by testing if one (or more) of the functions/objects exported from it exist. My main candidates are:

- `forwardRef`: but it might be discontinued as hinted here https://github.com/preactjs/preact/issues/2621
- `__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED`: I like that the name is pretty much guaranteed to be unique to this library, but at the same time might eventually be removed from React.
- `memo`: may not be unique enough.

Please what do you think? Do you have any suggestions? Thank you!

Reply:
```js
import { options } from 'preact';

let isCompat = false;

const oldHook = options.vnode;
options.vnode = vnode => {
  if (vnode['$$typeof']) isCompat = true;
  if (oldHook) oldHook(vnode);
}
```

This will detect `preact/compat` usage.

Reply:
Beautiful, thank you very much! Latest commit should now contain all the requested changes. :)